src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubListener.cs

```diff
@@ -12,180 +12,204 @@
 
 namespace Microsoft.Azure.WebJobs.ServiceBus
 {
-    // Created from the EventHubTrigger attribute to listen on the EventHub. 
     internal sealed class EventHubListener : IListener, IEventProcessorFactory
     {
         private readonly ITriggeredFunctionExecutor _executor;
-        private readonly EventProcessorHost _eventListener;
+        private readonly EventProcessorHost _eventProcessorHost;
         private readonly bool _singleDispatch;
         private readonly EventProcessorOptions _options;
         private readonly EventHubConfiguration _config;
         private bool _started;
 
-        public EventHubListener(ITriggeredFunctionExecutor executor, EventProcessorHost eventListener, bool single, EventHubConfiguration config)
+        public EventHubListener(ITriggeredFunctionExecutor executor, EventProcessorHost eventProcessorHost, bool singleDispatch, EventHubConfiguration config)
         {
-            this._executor = executor;
-            this._eventListener = eventListener;
-            this._singleDispatch = single;
-            this._options = config.GetOptions();
-            this._config = config;
+            _executor = executor;
+            _eventProcessorHost = eventProcessorHost;
+            _singleDispatch = singleDispatch;
+            _options = config.GetOptions();
+            _config = config;
         }
 
         void IListener.Cancel()
         {
             this.StopAsync(CancellationToken.None).Wait();
         }
 
-        void IDisposable.Dispose() // via IListener
+        void IDisposable.Dispose()
         {
-            // nothing to do. 
         }
 
-        // This will get called once when starting the JobHost. 
         public async Task StartAsync(CancellationToken cancellationToken)
         {
-            await _eventListener.RegisterEventProcessorFactoryAsync(this, _options);
+            await _eventProcessorHost.RegisterEventProcessorFactoryAsync(this, _options);
             _started = true;
         }
 
         public async Task StopAsync(CancellationToken cancellationToken)
         {
             if (_started)
             {
-                await _eventListener.UnregisterEventProcessorAsync();
+                await _eventProcessorHost.UnregisterEventProcessorAsync();
             }
             _started = false;
         }
-        
-        // This will get called per-partition. 
+         
         IEventProcessor IEventProcessorFactory.CreateEventProcessor(PartitionContext context)
         {
-            return new Listener(this);
+            return new EventProcessor(_config, new Checkpointer(), _executor, _singleDispatch);
         }
 
-        internal static Func<Func<Task>, Task> CreateCheckpointStrategy(int batchCheckpointFrequency)
```

Comment:
This javascript-like pattern and the way it was taking a closure was hard to understand. Simplifying it and making things more testable